src/Adapter/MSTest.TestAdapter/Helpers/RuntimeContext.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.Helpers;
+
+internal static class RuntimeContext
+{
+    // TODO: We should be using a capability from the runner instead of looking at environment variables.
+    public static bool IsInHotReloadContext()
```

Comment:
Do we expect this value to change over the lifetime of the process? Can you cache it?

Reply:
It shouldn't change so yeah I can cache it. Will do.